.travis.yml

```diff
@@ -143,13 +143,19 @@ before_install:
           (cd php-$MIN_PHP && ./configure --enable-sigchild --enable-pcntl && make -j2)
       fi
 
+    # install libsodium
+    - sudo add-apt-repository ppa:ondrej/php -y
+    - sudo apt-get update -q
+    - sudo apt-get install libsodium-dev -y
```

Comment:
I'd suggest to move these 3 lines below `Install extra PHP extensions`, in the same block (no dashes before)

Reply:
oh, and move them in the elif below, and wrap them in another if that checks if the `sodium.so` exists in the cache already or not